app/controllers/posts_controller.rb

```diff
@@ -6,6 +6,27 @@ def index
 
   def show
     @user = User.find(params[:user_id])
-    @post = Post.find(params[:id])
+    @post = @user.posts.find(params[:id])
+    @like = current_user.likes.where(post_id: params[:id]).exists?
+  end
+
+  def new
+    @current_user = current_user
+  end
```

Comment:
- [ ] I suggest you reimplement the logic for the new post action. There is no need for re-instantiating the current user since the curren_user is provided globally by the application controller. The new action should rather look like this:

  ```ruby
  def new
    @post = Post.new
  end
  ```

Reply:
Hey @Lameck1, Thank you for your feedback I appreciate it.
Forgive my ignorance please but if I don't set @current_user = current_user how can I pass the method's return to the method new.
I added @post = Post.new, though I don't know why to be honest.

Reply:
current_user is readily available since it is already implemented in the application controller. You only need to use it.

Reply:
There is no need to look for current user in the new action. It's not necessary.

Reply:
@Lameck1 Understood. I deleted it, thank you for coming back